game-app/game-core/src/main/java/games/strategy/triplea/ui/UnitChooser.java

```diff
@@ -89,6 +104,8 @@ private UnitChooser(
       final Collection<Unit> units,
       final CasualtyList defaultSelections,
       final Map<Unit, Collection<Unit>> dependent,
+      final boolean retreatPossibility,
+      final boolean movementForAirUnitsOnly,
```

Comment:
Thoughts on changing the parameters to accept a:
`UnitSeparator.SeparatorCategories` instead of adding additional parameters? I think that change would help avoid taking the parts of a `SeparatorCategories` to then simply assemble it when we could assemble it in advance, second the API to this method would be more usable.

Reply:
I inhereted this code change from Trevan. I am not familiar with the ```@Builder```-Pattern, i.e. I believe, I understand how   it works, but I have no experience with it and therefore no opinion how to use it. I am happy to follow your advice. (But I must say: Given the limited understanding I have of this, you haven't convinced me.)